src/App.tsx

```diff
@@ -2,24 +1,57 @@
+import React, { useState } from 'react';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const total = 42;
+const startingPage = 5;
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(startingPage);
+
+  const items = getNumbers(1, total);
+  const totalPages = Math.ceil(items.length / perPage);
+
+  const indexOfLastItem = currentPage * perPage;
+  const indexOfFirstItem = indexOfLastItem - perPage;
+  const currentItems = items.slice(indexOfFirstItem, indexOfLastItem);
```

Comment:
The calculation for pagination seems correct, but be aware that if 'currentPage' or 'perPage' are set to values that are not integers, this could lead to unexpected behavior. It might be a good idea to ensure these values are always rounded to the nearest integer.

Reply:
not the smallest possibility that these variables somehow would become not integers